### PR TITLE
refactor(CreatureAI): Use std::chrono for DoSummon()

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
-  Use std::chrono
-  

## Issues Addressed:

- ref https://github.com/azerothcore/azerothcore-wotlk/issues/6490

## Tests Performed:

- None

## How to Test the Changes:

1. Just check that despawntime is the same
2. for example check npc_earthmender_wilda

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Northrend/VioletHold/boss_ichoron.cpp [hunk 1]
```diff
@@ -193,8 +193,8 @@ class boss_ichoron : public CreatureScript
                             p1.m_positionY += 2.5f * sin(angle);
                             p2.m_positionX -= 2.5f * cos(angle);
                             p2.m_positionY -= 2.5f * sin(angle);
-                            DoSummon(NPC_ICHOR_GLOBULE, p1, 60000, TEMPSUMMON_TIMED_OR_DEAD_DESPAWN);
-                            DoSummon(NPC_ICHOR_GLOBULE, p2, 60000, TEMPSUMMON_TIMED_OR_DEAD_DESPAWN);
+                            DoSummon(NPC_ICHOR_GLOBULE, p1, 60s, TEMPSUMMON_TIMED_OR_DEAD_DESPAWN);
```
Comment (Winfidonarleyan) [inline src/server/scripts/Northrend/VioletHold/boss_ichoron.cpp:196]: Can change to 1m

**Comment by Bogir** [comment]
Time in "MINUTE" or "IN_MILLISECONDS" is better because it's easier  than used in Trinity std::chrono

**Comment by Winfidonarleyan** [comment]
> Time in "MINUTE" or "IN_MILLISECONDS" is better because it's easier than used in Trinity std::chrono

No

**Comment by Bogir** [comment]
> No

What easier?
`me->SummonCreature(entry, *me, TEMPSUMMON_TIMED_OR_CORPSE_DESPAWN, std::chrono::seconds(5));`
or
`me->SummonCreature(entry, *me, TEMPSUMMON_TIMED_OR_CORPSE_DESPAWN, 5000);`  or 5 * IN_MILLISECONDS
?
Why we need to complicate the code ?

**Comment by Kitzunu** [comment]
You dont do that, you do 5s

**Comment by Kitzunu** [comment]
@azerothcore/testers please test..

**Comment by Nyeriah** [comment]
Build is failing here:
```
/home/runner/work/azerothcore-wotlk/azerothcore-wotlk/src/server/game/AI/CreatureAI.cpp:362:55: fatal error: cannot convert 'Milliseconds' (aka 'duration<long, ratio<1, 1000> >') to 'uint32' (aka 'unsigned int') without a conversion operator
    return me->SummonCreature(entry, pos, summonType, static_cast<uint32>(despawnTime).count());
```

**Comment by Kitzunu** [comment]
Yep I know

**Comment by Nyeriah** [comment]
> Yep I know, have not had time to fix it

Okay 🙂 No worries, I'll test it once you check that so we can get things going.

**Review by Nyeriah — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
what's the status of this PR?

**Comment by Kitzunu** [comment]
need to fix buidl

**Comment by Kitzunu** [comment]
too much needs to be refactored

**Pull request closed without merging**